### PR TITLE
TIP-1171: Add events when adding, removing or changing the family of …

### PR DESCRIPTION
**Pull request by mmetayer**
…a product

**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Component/Product/Model/AbstractProduct.php [hunk 1]
```diff
@@ -47,9 +50,6 @@ abstract class AbstractProduct implements ProductInterface
     /** @var FamilyInterface $family */
     protected $family;
 
-    /** @var int */
-    protected $familyId;
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Component/Product/Model/AbstractProduct.php]: I took the opportunity to remove that :tada:

File: src/Akeneo/Pim/Enrichment/Component/Product/Model/Events/AddFamilyToProduct.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Enrichment\Component\Product\Model\Events;
+
+/**
+ * @author    Mathias METAYER <mathias.metayer@akeneo.com>
+ * @copyright 2019 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class AddFamilyToProduct
```
Comment (anaelChardan) [inline src/Akeneo/Pim/Enrichment/Component/Product/Model/Events/AddFamilyToProduct.php]: FamilyAddedToProduct, events must be at past tenses.

File: src/Akeneo/Pim/Enrichment/Component/Product/Model/Events/ChangeFamilyOfProduct.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Enrichment\Component\Product\Model\Events;
+
+/**
+ * @author    Mathias METAYER <mathias.metayer@akeneo.com>
+ * @copyright 2019 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ChangeFamilyOfProduct
```
Comment (anaelChardan) [inline src/Akeneo/Pim/Enrichment/Component/Product/Model/Events/ChangeFamilyOfProduct.php]: FamilyOfProductChanged

File: src/Akeneo/Pim/Enrichment/Component/Product/Model/Events/RemoveFamilyFromProduct.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Enrichment\Component\Product\Model\Events;
+
+/**
+ * @author    Mathias METAYER <mathias.metayer@akeneo.com>
+ * @copyright 2019 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class RemoveFamilyFromProduct
```
Comment (anaelChardan) [inline src/Akeneo/Pim/Enrichment/Component/Product/Model/Events/RemoveFamilyFromProduct.php]: FamilyRemovedFromProduct

**Pull request merged**